gnucash/import-export/ofx/gnc-ofx-import.cpp

```diff
@@ -1476,17 +1478,10 @@ void gnc_file_ofx_import (GtkWindow *parent)
             gnc_prefs_get_bool (GNC_PREFS_GROUP_IMPORT, GNC_PREF_AUTO_COMMODITY);
 
         DEBUG("Opening selected file(s)");
+
         // Create the structure that holds the list of files to process and the statement info.
-        info = new ofx_info;
-        info->num_trans_processed = 0;
-        info->statement = NULL;
-        info->last_investment_account = NULL;
-        info->last_import_account = NULL;
-        info->last_income_account = NULL;
-        info->parent = parent;
-        info->run_reconcile = FALSE;
-        info->file_list = selected_filenames;
-        info->response = 0;
+        auto info = new OfxInfo (parent, selected_filenames);
```

Comment:
If you make this a `std::unique_ptr` you won't have to worry about deleting it later, including worrying about there being a path that can escape the `delete`.

Reply:
I'm not sure how to write this as a unique pointer and get automatic memory management. Additionally the info must be sent to ofx calls.

Reply:
Replace `new OfxInfo` with `std::make_unique<OfxInfo>` to get a std::pointer. Use `info.get()` to get a plain `OfxInfo*` to hand to `ofx_set_foo_cb` to pass on to our libofx callbacks.

Reply:
see latest commit for attempt. `.get()` is ok for unique_ptr->raw_ptr but we'd also need the return trip.

Reply:
There is no return trip. The weird recursion has to be replaced with a loop in `gtk_file_osx_import`.